### PR TITLE
H-3082(I): Allow querying inheritance data from database

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

We store the inheritance data for data types in the database. This allows querying them.

## 🔍 What does this change?

- Move `DataTypeId` to `type-system`, this is required to allow the inheritance data to use the same types as the database
- Use `DataTypeId` instead of `VersionedUrl` when resolving types. 
- Add functionality to read inheritance data from the database when. As a proof-of-concept, this is used to read the existing inheritance data when inserting/updating data types.

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

### CODE REVIEW

File: tests/hash-graph-test-data/rust/src/data_type/object_v2.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "$id": "https://blockprotocol.org/@blockprotocol/types/data-type/object/v/2",
   "allOf": [
     {
-      "$ref": "https://blockprotocol.org/@blockprotocol/types/data-type/value/v/1"
+      "$ref": "https://blockprotocol.org/@blockprotocol/types/data-type/object/v/1"
```
Comment (TimDiekmann) [inline tests/hash-graph-test-data/rust/src/data_type/object_v2.json:7]: I don't know why this did not error before, but `object/v/2` is not on the whitelist of allowed primitives. To circumvent this, we just inherit from `object/v/1`.

File: apps/hash-graph/libs/graph/src/store/postgres/ontology/data_type.rs [hunk 2]
```diff
@@ -453,67 +490,60 @@ where
...
+            //   see https://linear.app/hash/issue/H-2976/validate-ontology-types-on-creation
+            // let closed_schema = data_type_validator
+            //     .validate(
+            //         ontology_type_resolver
+            //             .get_closed_data_type(*data_type_id)
+            //             .change_context(InsertionError)?,
+            //     )
+            //     .await
+            //     .attach(StatusCode::InvalidArgument)
+            //     .change_context(InsertionError)?;
```
Comment (TimDiekmann) [inline apps/hash-graph/libs/graph/src/store/postgres/ontology/data_type.rs:533]: This code did nothing before except pull all schemas from the cache. The validation basically was a no-op

File: apps/hash-graph/libs/graph/src/store/postgres/ontology/data_type.rs [hunk 3]
```diff
@@ -453,67 +490,60 @@ where
...
-                        limit: None,
-                        include_count: false,
-                    })
-                    .await
-                    .change_context(InsertionError)
-                    .attach_printable("Could not read parent data types")?
-                    .data_types,
-            )
-            .for_each(|data_type| {
-                ontology_type_resolver.add_open(Arc::new(data_type.schema));
```
Comment (TimDiekmann) [inline apps/hash-graph/libs/graph/src/store/postgres/ontology/data_type.rs:476]: We use the data from the database to calculate the inheritance chain for new types

File: tests/hash-graph-test-data/rust/src/data_type/object_v2.json (hunk 1 above)
Comment (TimDiekmann) [reply tests/hash-graph-test-data/rust/src/data_type/object_v2.json]: Fixed in #5308

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**